# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1463424860 25200
#      Mon May 16 11:54:20 2016 -0700
# Node ID 09e1bb792dd976f8f2af11691b0c8e17161e869f
# Parent  2e0e4eec0d62475ab9d531b14b09d7327e1e8081
Fix compilation error when building without sndfile

* audioread.cc (audio_sub_functions): Conditionally compile when HAVE_SNDFILE
is defined.

diff --git a/libinterp/dldfcn/audioread.cc b/libinterp/dldfcn/audioread.cc
--- a/libinterp/dldfcn/audioread.cc
+++ b/libinterp/dldfcn/audioread.cc
@@ -514,16 +514,18 @@ Return information about an audio file s
   octave_unused_parameter (args);
 
   err_disabled_feature ("audioinfo",
                         "reading and writing sound files through libsndfile");
 
 #endif
 }
 
+#ifdef HAVE_SNDFILE
+
 static void
 audio_sub_formats (int format)
 {
   int count;
   sf_command (NULL, SFC_GET_FORMAT_SUBTYPE_COUNT, &count, sizeof (int));
 
   for (int i = 0; i < count; i++)
     {
@@ -536,16 +538,18 @@ audio_sub_formats (int format)
       sfinfo.channels = 1;
       sfinfo.format = (format & SF_FORMAT_TYPEMASK) | info.format;
 
       if (sf_format_check (&sfinfo))
         octave_stdout << "  " << info.name << std::endl;
     }
 }
 
+#endif
+
 DEFUN_DLD (audioformats, args, ,
 "-*- texinfo -*-\n\
 @deftypefn  {} {} audioformats ()\n\
 @deftypefnx {} {} audioformats (@var{format})\n\
 Display information about all supported audio formats.\n\
 \n\
 If the optional argument @var{format} is given, then display only formats\n\
 with names that start with @var{format}.\n\
